### DISCUSSION TITLE
Navigating through history of a line (or few)

### DISCUSSION THREAD

**Discussion by rkouznetsov** [tpope/vim-fugitive#1862] [Q&A] [answered]
Hi,

I would want to see how a specific piece of code (e.g. line or few) were revised in past.
Now i do Gblame, Gdiff to the revision were the piece was last changed, Gblame there,  check a revision of the previous change Gdiff to that revision etc... I believe this process could be automated.
I wonder if there is a way to make vsplit (like in Gdiff), and then order to the left tab a revision with next/previous state  of a selected line range with some hotkey. It would be great to have such a tool if it does not exist already...

Thank you!

**Comment by tpope** [comment, marked as answer]
This is already fairly automated:

* Press return in the blame window to bring up the diff. The appropriate `+` line will be selected.
* Find the appropriate `-` line and press return again to open up the file at the commit.
* Press return one more time to blame again.
* Repeat.

**Reply by rkouznetsov** [reply]
Thank you! Very useful. I have yet to figure out and get used to how the thing interacts with diff-tabs...
